beacon-chain/p2p/pubsub.go

```diff
@@ -144,6 +173,43 @@ func msgIDFunction(pmsg *pubsub_pb.Message) string {
 	return string(h[:20])
 }
 
+// Spec:
+// The derivation of the message-id has changed starting with Altair to incorporate the message topic along with the message data.
+// These are fields of the Message Protobuf, and interpreted as empty byte strings if missing. The message-id MUST be the following
+// 20 byte value computed from the message:
+//
+// If message.data has a valid snappy decompression, set message-id to the first 20 bytes of the SHA256 hash of the concatenation of
+// the following data: MESSAGE_DOMAIN_VALID_SNAPPY, the length of the topic byte string (encoded as little-endian uint64), the topic
+// byte string, and the snappy decompressed message data: i.e. SHA256(MESSAGE_DOMAIN_VALID_SNAPPY + uint_to_bytes(uint64(len(message.topic)))
+// + message.topic + snappy_decompress(message.data))[:20]. Otherwise, set message-id to the first 20 bytes of the SHA256 hash of the concatenation
+// of the following data: MESSAGE_DOMAIN_INVALID_SNAPPY, the length of the topic byte string (encoded as little-endian uint64),
+// the topic byte string, and the raw message data: i.e. SHA256(MESSAGE_DOMAIN_INVALID_SNAPPY + uint_to_bytes(uint64(len(message.topic))) + message.topic + message.data)[:20].
+func (s *Service) altairMsgID(pmsg *pubsub_pb.Message) string {
+	topic := *pmsg.Topic
+	topicLen := uint64(len(topic))
+	topicLenBytes := bytesutil.Uint64ToBytesLittleEndian(topicLen)
+
+	decodedData, err := encoder.DecodeSnappy(pmsg.Data, params.BeaconNetworkConfig().GossipMaxSize)
+	if err != nil {
```

Comment:
I wonder if we should handle this error, even just for informational purposes to know it happened with a debug log or log.WithError(err).Debug

Reply:
This would basically blow up our debug logs, If you are on a noisy subnet you will have a burst of 200 messages every slot, which would make it difficult to parse logs. In that case, having non snappy compliant messages is still accepted, so I don't think it is fully necessary for logging it.